tests/datadriven/__init__.py

```diff
@@ -194,6 +200,24 @@ def assertRaisesWith(self, exceptionType):
                 "exception of type {} not raised".format(
                     exceptionType.__name__))
 
+    def assertAlmostEqual(self, a, b, ndigits=7):
+        """
+        Assert that a and b are equal within ndigits digits
+        """
+        if round(a-b, ndigits) != 0:
```

Comment:
I'm worried that this implementation doesn't actually test the equality to within n significant digits. Consider something like `assertAlmostEqual(1.111111111E20, 1.111111112E20, 5)`, which should, to my mind, return True, but will return False in this implementation. I suggest something like:
```
if b!= 0 and round(abs((a-b)/b), ndigits) != 0
```
(handle case where `b == 0` separately)
I looked around to see if there was an easy way to just get the mantissa of a floating point number in python to avoid doing the division and couldn't find anything simple and consistent.